src/components/pagination/Pagination.vue

```diff
@@ -0,0 +1,125 @@
+          {{ " " }}
+          {{ $t("results") }}
+        </p>
+      </div>
+      <div>
+        <nav
+          class="isolate inline-flex -space-x-px rounded-md shadow-sm"
+          aria-label="Pagination"
+        >
+          <a
+            href="#"
+            class="relative inline-flex items-center rounded-l-md px-2 py-2 text-gray-400 ring-1 ring-inset ring-gray-300 hover:bg-gray-50 focus:z-20 focus:outline-offset-0"
+          >
+            <span class="sr-only">Previous</span>
+            <ChevronLeftIcon
+              class="h-5 w-5"
+              aria-hidden="true"
+              @click="previousPage"
+            />
+          </a>
+          <div>
+            <a
+              v-for="pageIndex in totalPages"
+              :key="pageIndex"
+              href="#"
+              :class="{
+                'bg-indigo-600 text-white': page === pageIndex,
+                'text-gray-400': page !== pageIndex,
+              }"
+              class="relative z-10 inline-flex items-center px-4 py-2 text-sm font-semibold focus:z-20 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
+              @click="changePage(pageIndex)"
+              >{{ pageIndex }}</a
+            >
+          </div>
+
+          <a
+            href="#"
+            class="relative inline-flex items-center rounded-r-md px-2 py-2 text-gray-400 ring-1 ring-inset ring-gray-300 hover:bg-gray-50 focus:z-20 focus:outline-offset-0"
+          >
+            <span class="sr-only">Next</span>
+            <ChevronRightIcon
+              class="h-5 w-5"
+              aria-hidden="true"
+              @click="nextPage"
+            />
+          </a>
+        </nav>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { usePostStore } from "@/modules/auth/store/postStore";
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+} from "@zhuowenli/vue-feather-icons";
+import { computed, ref } from "vue";
+
+const { totalPosts } = defineProps({
+  totalPosts: {
+    type: Number,
+    default: 0,
+  },
+});
+
+const postStore = usePostStore();
+
+const page = ref(parseInt(localStorage.getItem("currentPage")) || 1);
+const limit = computed(() => postStore.posts.limit || 30);
+const totalPages = computed(() => Math.ceil(totalPosts / limit.value) || 5);
+
+function changePage(newPage: number) {
+  page.value = newPage;
+  localStorage.setItem("currentPage", page.value.toString());
+  postStore.getPostsByPagination(page.value, limit.value);
+}
+
+function nextPage() {
+  if (page.value < totalPages.value) {
+    page.value += 1;
+    localStorage.setItem("currentPage", page.value.toString());
+    postStore.getPostsByPagination(page.value, limit.value);
+  }
+}
+
+function previousPage() {
+  if (page.value > 1) {
+    page.value -= 1;
+    localStorage.setItem("currentPage", page.value.toString());
+    postStore.getPostsByPagination(page.value, limit.value);
+  }
+}
```

Comment:
why not call changePage(page.value - 1) function instead of nextPage and previosPage, you have 3 functions that do almost the same thing